src/cmap/auth/mongo_credentials.ts

```diff
@@ -132,15 +132,15 @@ export class MongoCredentials {
     ) {
       if (this.source != null && this.source !== '$external') {
         // TODO(NODE-3483): Replace this with a MongoAuthValidationError
-        throw new MongoDriverError(
+        throw new MongoRuntimeError(
```

Comment:
Is the `validate()` function only used internally? Feels like the errors here could be API, not runtime

Reply:
I think these are all API, this function (I believe) is only called during options parsing so it would be sometime identified at startup essentially